src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -1142,6 +1142,24 @@ export class Workbench implements IPartService {
 		}
 	}
 
+	// Resize requested part along the main axis
+	// layout will do all the math for us and adjusts the other Parts
+	public resizePart(part: Parts, sizeChangePx: number, skipLayout?: boolean): TPromise<void> {
```

Comment:
This method does not need to return a promise and also does not seem to be using `skipLayout`. I would suggest to do the `layout()` call from within `layout.ts` and not outside.